### PR TITLE
feat: add flag to force push repositories to gitea during deployment

### LINKED ISSUE #1410: Support force pushing to repositories using the Refspec format
### Is your feature request related to a problem? Please describe.
Now that Zarf supports git refspecs we should support the + prefix to allow force push behavior to be specified.

### Describe the solution you'd like
Using the + syntax of git refspecs to enable force pushing of git repos in the airgap on Zarf deploy.  (Note this will need to be detected and done both for go-git and the git fallback).

The changes should be primarily here: https://github.com/defenseunicorns/zarf/blob/git-kiss/src/internal/packager/git/push.go along with relevant docs: https://github.com/defenseunicorns/zarf/tree/git-kiss/examples/git-data and tests: https://github.com/defenseunicorns/zarf/blob/git-kiss/src/test/e2e/22_git_and_flux_test.go

### Describe alternatives you've considered
We could use URI syntax to introduce something like `?force=true` but this may be confusing for those familiar with `git` and may be misleading since that never is passed to an http server.

### Additional context
This is a continuation of the work done in https://github.com/defenseunicorns/zarf/pull/1404

### PR DESCRIPTION
**Pull request by Jneville0815**
## Description

Add flag to force push repositories to gitea during deployment. Not sure if any tests need to be created or updated for this change?
...

## Related Issue

Fixes #1410 

Relates to #1410

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/zarf-dev/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

**Comment by phillebaba** [comment]
@Jneville0815 could you generate docs and then we can merge.

**Comment by Jneville0815** [comment]
@phillebaba done

**Comment by Jneville0815** [comment]
@phillebaba any update on this?

File: src/cmd/package.go [hunk 1]
```diff
@@ -551,6 +552,7 @@ func bindDeployFlags(v *viper.Viper) {
 	deployFlags.StringVar(&pkgConfig.PkgOpts.Shasum, "shasum", v.GetString(common.VPkgDeployShasum), lang.CmdPackageDeployFlagShasum)
 	deployFlags.StringVar(&pkgConfig.PkgOpts.SGetKeyPath, "sget", v.GetString(common.VPkgDeploySget), lang.CmdPackageDeployFlagSget)
 	deployFlags.BoolVar(&pkgConfig.PkgOpts.SkipSignatureValidation, "skip-signature-validation", false, lang.CmdPackageFlagSkipSignatureValidation)
+	deployFlags.BoolVar(&pkgConfig.DeployOpts.ForcePushRepos, "force-push-repos", false, lang.CmdPackageForcePushRepos)
```
Comment (phillebaba) [inline src/cmd/package.go:555]: This also has to be added to the mirror command flags.

**Pull request closed without merging**